core/dbt/graph/selector_methods.py

```diff
@@ -165,8 +221,111 @@ def search(self, included_nodes, selector):
                 yield node
 
 
-class MethodName(StrEnum):
-    FQN = 'fqn'
-    Tag = 'tag'
-    Source = 'source'
-    Path = 'path'
+class PackageSelectorMethod(SelectorMethod):
+    def search(
+        self, included_nodes: Set[UniqueId], selector: str
+    ) -> Iterator[UniqueId]:
+        """Yields nodes from included that have the specified package"""
+        for node, real_node in self.all_nodes(included_nodes):
+            if real_node.package_name == selector:
+                yield node
+
+
+def _getattr_descend(obj: Any, attrs: List[str]) -> Any:
+    value = obj
+    for attr in attrs:
+        value = getattr(value, attr)
+    return value
+
+
+class CaseInsensitive(str):
+    def __eq__(self, other):
+        if isinstance(other, str):
+            return self.upper() == other.upper()
+        else:
+            return self.upper() == other
+
+
+class ConfigSelectorMethod(SelectorMethod):
+    def search(
+        self, included_nodes: Set[UniqueId], selector: str
+    ) -> Iterator[UniqueId]:
+        parts = self.arguments
```

Comment:
This means _all_ config arguments should be fair game for `'config.arg:value'`, right?

These worked for me on Fishtown's internal analytics project:
```
dbt run -m 'config.schema:analytics_utils'
dbt run -m 'config.materialized'
dbt test -m 'config.severity:error'
```

Whereas these did not match any nodes, when I believe they should have:
```
dbt run -m 'config.incremental_strategy:delete+insert'
dbt run -m 'config.unique_key:event_id'
```

Reply:
Oh, nice catch! That's because `unique_key` and `incremental_strategy` are special, so they don't show up to `getattr`. I'll change this to treat things like dictionaries when `getattr` fails.